lib/galaxy/tool_util/lint.py

```diff
@@ -12,15 +12,36 @@
 
 
 def lint_tool_source(tool_source, level=LEVEL_ALL, fail_level=LEVEL_WARN, extra_modules=None, skip_types=None, name=None):
+    """
+    apply all (applicable) linters from the linters submodule
+    and the ones in extramodules
+
+    immediately print linter messages (wrt level) and return if linting failed (wrt fail_level)
+    """
     extra_modules = extra_modules or []
     skip_types = skip_types or []
     lint_context = LintContext(level=level, skip_types=skip_types, object_name=name)
     lint_tool_source_with(lint_context, tool_source, extra_modules)
-
     return not lint_context.failed(fail_level)
 
 
+def lint_context_for_tool_source(tool_source, extra_modules=None, skip_types=None, name=None):
```

Comment:
I would probably change the signature a bit like this to be more explicit about what we are returning. But for that, we need to move the declaration of LintContext and LintMessage before this or use forward declaration like `-> "LintContext"`

```suggestion
def get_lint_context_for_tool_source(tool_source, extra_modules=None, skip_types=None, name=None) -> LintContext:
```